enhancements/agent-installer/development-environment.md

```diff
@@ -0,0 +1,305 @@
+  - https://issues.redhat.com/browse/AGENT-20
+---
+
+# Ephemeral Agent Installer Development Infrastructure
+
+## Release Signoff Checklist
+
+- [ ] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Operational readiness criteria is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+## Summary
+
+Create a development environment for the ephemeral agent based installer image in the 
+[dev-scripts](https://github.com/openshift-metal3/dev-scripts) repository and refactor 
+components as needed for the new project. In addition we will add base case e2e test flows
+integrated into Openshift CI to get the project started.
+
+## Motivation
+
+The new agent based installer project will need a standard development environment for
+the team to use. We will need an e2e testing framework for validation in Openshift CI. 
+In this enhancement we will explain our desire to reuse existing frameworks for the new 
+project while limiting the scope of modifications to them to the new project's requirements.
+
+In this space we have two existing frameworks that could be used for this new project. These 
+frameworks are:
+* [dev-scripts](https://github.com/openshift-metal3/dev-scripts): Created by the Metal Platform 
+(IPI) team to set up environments to install OCP both for local development and in Openshift CI.
+* [assisted-test-infra](https://github.com/openshift/assisted-test-infra): Created by the Assisted
+Installer team to run full e2e test flows.
+
+Looking at existing e2e test flow steps from these two frameworks it was determined that 
+in dev-scripts test flows the beginning steps are very similar to the development
+flows we desire for the agent installer. In contrast the assisted-test-infra test flows
+set up extra things in the development environment that is undesirable for the agent installer.
+
+### Goals
+
+* Using this development environment in the new ephemeral agent based installer image project.
+* Re-factoring parts of dev-scripts that are needed for this ephemeral agent based installer.
+* Convergence of the underlying scripts and environment setup code between Metal Platform (IPI) 
+team and the new Agent Installer teams.
+
+### Non-Goals
+
+* Re-factoring dev-scripts and assisted-test-infra for reasons not related to the new 
+ephemeral agent based installer image project.
+* Overall convergence between all the installer development teams of scripts and e2e test flows.
+
+## Proposal
+
+Create an development environment for the ephemeral agent based installer image in the 
+[dev-scripts](https://github.com/openshift-metal3/dev-scripts) repository and refactor 
+components as needed for the new project.
+
+### User Stories
+
+* As a developer, I need to set up an environment so that I can simulate the installation of a 
```

Comment:
I'd say something like "emulate on-premise customer environments (including various network and server configurations), to test installation using the ephemeral agent based installer"

Reply:
I'll put in the adjective change here. :) 

But, just to give some context the use of simulate vs emulate is probably because of my hardware background. Hardware emulation vs Software simulation is the context I'm used to having these words. In that context these are pretty different things despite both being software programs. What we are doing with the VM environment for installation is close to "a software based simulation of a customers installation environment." That's how I would describe it... but that isn't too far off from Steve's wording either.

Reply:
It's a real installation into a simulated environment (VMs and linux bridges instead of servers and switches).